addons/stock/models/stock_production_lot.py

```diff
@@ -86,7 +86,7 @@ def write(self, vals):
     def _product_qty(self):
         for lot in self:
             # We only care for the quants in internal or transit locations.
-            quants = lot.quant_ids.filtered(lambda q: q.location_id.usage in ['internal', 'transit'])
+            quants = lot.quant_ids.filtered(lambda q: q.location_id.usage == "internal" or (q.location_id.usage == "transit" and q.location_id.company_id))
```

Comment:
``
q.location_id.company_id == lot.company_id
``

Reply:
Simon and I think only check the existence of location_id.company_id is enough, since lot with a different company_id can't be put in that location. What do you think?